src/Orleans/Utils/IAsymmetricComparer.cs

```diff
@@ -0,0 +1,16 @@
+﻿
+namespace Orleans.Runtime
+{
+    public interface IAsymmetricComparer<in T1, in T2>
```

Comment:
You think this justifies a public class in a general **public** Orleans utils? 
We used to have much a higher bar for releasing any general purpose public utility. Basically almost nothing. @sergeybykov was usually a bit proponent of releasing any public general purpose .NET utility. Almost all classes in this folder are internal classes.